pandas/_libs/__init__.py

```diff
@@ -10,6 +10,10 @@
 ]
 
 
+# Below import needs to happen first to ensure pandas top level
+# module gets monkeypatched with the pandas_datetime_CAPI
+# see pandas_datetime_exec in pd_datetime.c
+import pandas._libs.pandas_datetime as pandas_datetime  # noqa # isort: skip
```

Comment:
Not a fan of this, but capsules currently cannot be attached to anything but a top level package. See https://github.com/python/cpython/pull/6898 which I think would solve this, but for now this has to be attached to the top level namespace and really come before anything else to avoid circular import errors